pkg/http/server.go

```diff
@@ -62,8 +86,96 @@ type Server struct {
 	requestCount           int32                               // counter holds no. of request in process.
 }
 
+// certificate securely updates server with a new certificate
+func (srv *Server) certificate() (*tls.Certificate, error) {
+	srv.certMu.Lock()
+	defer srv.certMu.Unlock()
+	return srv.currCert, nil
+}
+
+// updateCertOCSP updates the OCSP field of the current certificate
+func (srv *Server) updateCertOCSP() (bool, bool, error) {
+
+	srv.certMu.Lock()
+	publicCert := srv.currCert.Certificate[0]
+	srv.certMu.Unlock()
+
+	// Extract the raw bytes of the leaf public certificate
+	parsedCert, err := x509.ParseCertificate(publicCert)
+	if err != nil {
+		return false, false, err
+	}
+
+	// Fetch OCSP response
+	ocspBytes, ocspResp, ocspErr := ocsp.GetOCSPForCert([]*x509.Certificate{parsedCert})
+	if ocspErr != nil {
+		// Return OCSP unsupported
+		if ocspErr == ocsp.ErrNoOCSPServer {
+			return false, true, nil
+		}
+		return false, false, ocspErr
+	}
+
+	srv.certMu.Lock()
+	srv.currCert = &tls.Certificate{
+		Certificate: srv.currCert.Certificate,
+		PrivateKey:  srv.currCert.PrivateKey,
+		OCSPStaple:  ocspBytes,
+	}
+	srv.certMu.Unlock()
+
+	if ocspResp.Status == ocspgo.Revoked {
+		return false, false, ErrCertRevoked
+	}
+
+	return true, false, nil
+}
+
+// startOCSPBackgroundUpdate - regularly updates tls certificate with
+// new ocsp staple response.
+func (srv *Server) startOCSPBackgroundUpdate() {
```

Comment:
As I remember and my understanding is that we agreed to have OCSP update in `minio` not in `http`.  Am I missed something here?